app/controllers/concerns/sorting.rb

```diff
@@ -8,7 +8,14 @@ module Sorting
 
   included do
     def build_order_by(model, *fields)
-      fields.flatten.each_with_object({}) do |field, obj|
+      order = build_fields_order_by(model, *fields)
+      # add id for deterministic pagination
+      order[:id] = :asc unless order.include?(:id)
```

Comment:
You can try commenting this line and running those newly added tests.

Reply:
What about something like:
```ruby
order.merge(id: :asc) unless order.include(:id)
```

Then you can probably omit the the last line

Reply:
Or even better :rocket: :sunglasses: 
```ruby
order.reverse_merge(id: :asc)
```

Reply:
Withot allocation:
```ruby
build_fields_order_by(model, *fields).reverse_merge(id: :asc)
```

...just thinking out loud as I feel lonely at the office, please come @vkrizan ... :sob: :laughing:

Reply:
`reverse_merge` would put `id` at the start, however it requires to have the id last.